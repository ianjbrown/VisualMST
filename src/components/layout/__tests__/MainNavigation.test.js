import { render, screen, cleanup } from "@testing-library/react";
import MainNavigaton from "../MainNavigation";
import { BrowserRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom/";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Test Navigation Bar matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <MainNavigaton />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

const links = [
  { text: "Home", location: "/" },
  { text: "About", location: "/about" },
  { text: "Contact", location: "/contact" },
  { text: "Algorithms", location: "/algorithms" },
];

test.each(links)("Check if Navigation Bar has %s link", (link) => {
  render(
    <Router>
      <MainNavigaton />
    </Router>
  );
  const linkDOM = screen.getByText(link.text);
  expect(linkDOM).toHaveAttribute("href", link.location);
});
