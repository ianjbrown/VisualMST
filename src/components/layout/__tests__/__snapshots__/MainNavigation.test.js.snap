// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation Bar matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      VisualMST
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="active nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/about"
          disabled={false}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/algorithms"
          disabled={false}
          href="/algorithms"
          onClick={[Function]}
        >
          Algorithms
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/contact"
          disabled={false}
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </div>
    </div>
  </div>
</nav>
`;
